### ISSUE TITLE
About batch_size

### ISSUE THREAD

**Issue by Yan1026** [yyliu01/PS-MT#14]
HI! I train with `./scripts/train_voc_aug.sh -l 1323 -g 2 -b 101 `，and I use `4*V100(16G),batchsize=4,lr=0.0025`.  In your code,it's `2*V100(32G),batchsize=8,lr=0.0025`.But miou in result is just75.29%.When I set lr=0.00125 ,the miou=76.33%,lower than you 78.20%.

I set batchsize in scripts/train_voc_aug.sh line 49：
```
nohup python3 VocCode/main.py --labeled_examples="${labelled}" --gpus=${gpus} --backbone=${backbone} --warm_up=5 --batch_size=4 --semi_p_th=.6 --semi_n_th=.0 --learning-rate=1.25e-3 \
--epochs=${max_epochs} --unsup_weight=${unsup_weight} > voc_aug_"${labelled}"_"${backbone}".out &
```

In addition, I have done experiments on Cityscapes(372), and only with batchsize set to 2 does not get error:out of GUP memory.`(gpu=4*v100(16G),lr=0.0045)`.In your code it's `gpu=2*v100(32G),lr=0.0045,batchsize=8`

Can you give me some advice?

**Comment by yyliu01** [comment]
Hi @Yan1026 ,

I recommend you follow the setting in this link (https://wandb.ai/pyedog1976/PS-MT(VOC12)/runs/177s76t6) for voc12 aug set and this link (https://wandb.ai/pyedog1976/PS-MT(City)/runs/bxf92y1q) for cityscapes.

In case you are using V100 (16G), I suggest you to downgrade backbone to be resnet50 and compare the results in paper.

**Comment by Yan1026** [comment]
Thanks for reply,I will try again.
I just set batchsize in scripts/train_voc_aug.sh line 49：
`nohup python3 VocCode/main.py --labeled_examples="${labelled}" --gpus=${gpus} --backbone=${backbone} --warm_up=5 --batch_size=4 --semi_p_th=.6 --semi_n_th=.0 --learning-rate=1.25e-3 \
--epochs=${max_epochs} --unsup_weight=${unsup_weight} > voc_aug_"${labelled}"_"${backbone}".out &`

Is this right?

**Comment by yyliu01** [comment]
Hi @Yan1026 , 

Sorry, I can't guarantee the performance if your setting differs from the hype-parameter in this link (https://wandb.ai/pyedog1976/PS-MT(VOC12)/runs/177s76t6/overview?workspace=user-pyedog1976). 

The smaller batch size or learning rate will lead to a significant difference, and please follows our setting for the aug experiment.

**Comment by JoyHuYY1412** [comment]
> Thanks for reply,I will try again. I just set batchsize in scripts/train_voc_aug.sh line 49： `nohup python3 VocCode/main.py --labeled_examples="${labelled}" --gpus=${gpus} --backbone=${backbone} --warm_up=5 --batch_size=4 --semi_p_th=.6 --semi_n_th=.0 --learning-rate=1.25e-3 \ --epochs=${max_epochs} --unsup_weight=${unsup_weight} > voc_aug_"${labelled}"_"${backbone}".out &`
> 
> Is this right?

Hi Yuyuan, thanks for your great work. I also have one question about your batchsize. You mentioned that `On Pascal VOC 2012, we crop images to 512 × 512 pixels for DeeplabV3+, train for 80 epochs with base learning rate set to 0.01, using batch size 32, for both labelled and unlabelled images, following CPS`. So you are using batchsize=64, am I right? However, it seems tp me that CPS only use batchsize=16 (I am not sure about that). Could you give me some advice on that? Thank you so much.
